test/DevicesController.test.js

```diff
@@ -0,0 +1,139 @@
+import test from 'ava';
+import request from 'supertest-as-promised';
+import ouathClients from '../src/oauthClients.json';
+import app from './testApp';
+import settings from './settings';
+
+const USER_CREDENTIALS = {
+  password: 'password',
+  username: 'deviceTestUser@test.com',
+};
+
+const DEVICE_ID = '350023001951353337343732';
+const TEST_PUBLIC_KEY =
+  '-----BEGIN PUBLIC KEY-----\n' +
+  'MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCsxJFqlUOxK5bsEfTtBCe9sXBa' +
+  '43q9QoSPFXEG5qY/+udOpf2SKacgfUVdUbK4WOkLou7FQ+DffpwztBk5fWM9qfzF' +
+  'EQRVMS8xwS4JqqD7slXwuPWFpS9SGy9kLNy/pl1dtGm556wVX431Dg7UBKiXuNGR' +
+  '7E8d2hfgeyiTtsWfUQIDAQAB\n' +
+  '-----END PUBLIC KEY-----\n';
+
+let testUser;
+let userToken;
+let deviceToApiAttributes;
+
+test.before(async () => {
+  const userResponse = await request(app)
+    .post('/v1/users')
+    .send(USER_CREDENTIALS);
+
+  testUser = userResponse.body;
+
+  const tokenResponse = await request(app)
+    .post('/oauth/token')
+    .set('Content-Type', 'application/x-www-form-urlencoded')
+    .send({
+      client_id: ouathClients[0].clientId,
+      client_secret: ouathClients[0].clientSecret,
+      grant_type: 'password',
+      password: USER_CREDENTIALS.password,
+      username: USER_CREDENTIALS.username,
+    });
+
+  userToken = tokenResponse.body.access_token;
+});
+
+// TODO come up better test name
+test.serial('provision test', async t => {
```

Comment:
can't find a good name for the test, it tests provision endpoint.
Btw I decided to test provisionController here because the endpoint creates a device for all other tests.